src/ax3/filters/CoerceToBool.hx

```diff
@@ -27,7 +27,7 @@ class CoerceToBool extends AbstractFilter {
 			case TTBoolean:
 				e; // shouldn't happen really
 
-			case TTFunction | TTFun(_) | TTClass | TTObject(_) | TTInst(_) | TTStatic(_) | TTArray(_) | TTVector(_) | TTRegExp | TTXML | TTXMLList | TTDictionary(_, _):
```

Comment:
Why tho? Is this about empty strings typed as objects?

Reply:
This here is about having a boolean stored in an Object and coercing it back to boolean. It's a bit unfortunate again as we have some abstract data storage class, that is later used with the proper type, in this case boolean. Similar to this:

```
class A {
    protected var _data:Object;
}

class B extends A {
    public function getData():boolean {
        return _data;
    }
    public function setData(value:boolean) {
        _data = value;
    }
}
```

It converted to:

```
public function getData():Bool {
    return _data != null;
}
```

but that's not correct as AS3 will coerce to boolean at run-time, similar to what we already have for ASAny. Now ASAny and ASObject behave the same when coercing to Bool.

Reply:
Right. Sigh :)